cmd/world/root/login.go

```diff
@@ -0,0 +1,183 @@
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			logger.SetDebugMode(cmd)
+
+			err := loginOnBrowser(cmd.Context())
+			if err != nil {
+				return eris.Wrap(err, "failed to login")
+			}
+
+			return nil
+		},
+	}
+
+	return loginCmd
+}
+
+func loginOnBrowser(ctx context.Context) error {
+	encryption, err := login.NewEncryption()
+	if err != nil {
+		logger.Error("Failed to create login encryption", err)
+		return err
+	}
+
+	encodedPubKey := encryption.EncodedPublicKey()
+	sessionID := uuid.NewString()
+	tokenName := generateTokenNameWithFallback()
+
+	loginURL := fmt.Sprintf("%s/cli/login?session_id=%s&token=%s&pub_key=%s",
+		worldForgeBaseURL, sessionID, tokenName, encodedPubKey)
+
+	loginMessage := "In case the browser didn't open, please open the following link in your browser"
+	fmt.Print(style.CLIHeader("World Forge", style.DoubleRightIcon.Render(loginMessage)), "\n")
+	fmt.Printf("%s\n\n", loginURL)
+	if err := login.RunOpenCmd(ctx, loginURL); err != nil {
+		logger.Error("Failed to open browser", err)
+		return err
+	}
+
+	// Wait for the token to be generated
+	if err := program.RunProgram(ctx, func(p program.Program, ctx context.Context) error {
+		p.Send(program.StatusMsg("Waiting response from world forge service..."))
+
+		pollURL := fmt.Sprintf("%s/auth/cli/login/%s", worldForgeBaseURL, sessionID)
+		accessToken, err := pollForAccessToken(ctx, pollURL)
+
+		if err != nil {
+			return err
+		}
+
+		token, err = encryption.DecryptAccessToken(accessToken.AccessToken, accessToken.PublicKey, accessToken.Nonce)
+		if err != nil {
+			return err
+		}
+
+		if err := globalconfig.SetWorldForgeToken(tokenName, token); err != nil {
+			logger.Error("Failed to set access token", err)
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		logger.Error("Failed to get access token", err)
+		return err
+	}
+
+	fmt.Println(style.TickIcon.Render("Successfully logged in :"))
+	// Print the token
+	credential, err := globalconfig.GetWorldForgeCredential()
+	if err != nil {
+		logger.Warn("Failed to get the access token when print", err)
+	}
+	stringCredential, err := json.MarshalIndent(credential, "", "  ")
+	if err != nil {
+		logger.Warn("Failed to marshal the access token when print", err)
+	}
+	fmt.Println(style.BoldText.Render(string(stringCredential)))
+	return nil
+}
+
+func generateTokenName() (string, error) {
+	user, err := user.Current()
+	if err != nil {
+		return "", eris.Wrap(err, "cannot retrieve current user")
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return "", eris.Wrap(err, "cannot retrieve hostname")
+	}
+
+	return fmt.Sprintf("cli_%s@%s_%d", user.Username, hostname, time.Now().Unix()), nil
+}
+
+func generateTokenNameWithFallback() string {
+	name, err := generateTokenName()
+	if err != nil {
+		name = fmt.Sprintf("cli_%d", time.Now().Unix())
+	}
+	return name
+}
+
+func pollForAccessToken(ctx context.Context, url string) (login.AccessTokenResponse, error) {
```

Comment:
This might be a DOS vulnerability source(?). We should probably come up with rules on how many login attempts a user can do in a given time, e.g. max 3 login attempts in 1 minute, to not overload the server with requests. We can also set the user-agent of the request to something like 'world-cli', and reject all http requests without it. We can ignore this for now. created a [ticket for this](https://linear.app/arguslabs/issue/WORLD-1138/increase-world-forge-security)